src/validators/model.rs

```diff
@@ -157,6 +159,18 @@ impl Validator for ModelValidator {
                 input,
             ))
         } else {
+            if self.custom_init {
+                // If we wanted, we could introspect the __init__ signature, and store the
+                // keyword arguments and types, and create a validator for them.
+                // Perhaps something similar to `validate_call`? Could probably make
+                // this work with from_attributes, and would essentially allow you to
+                // handle init vars by adding them to the __init__ signature.
+                // TODO: Should probably provide some nicer error if not a dict, though if I
+                //   understand correctly, this current logic actually seems to match v1 behavior..
+                if let Ok(py_dict) = input.to_object(py).downcast::<PyDict>(py) {
```

Comment:
I've prefer a dedicated method on `input`, then custom implementations for python and JSON.

Python is just `.downcast()`, but with JSON we only want to cast the whole thing to python if it is indeed an object.

Reply:
I fiddled longer with this than it probably made sense to.

Tried to do something like this on input:
```rust
    fn to_py_dict(&self, py: Python<'a>) -> Option<&'a PyDict> {
        self.to_object(py).downcast::<PyDict>(py).ok()
    }
```
and was getting this error:
```
error[E0515]: cannot return value referencing temporary value
   --> src/input/input_abstract.rs:276:9
    |
276 |         self.to_object(py).downcast::<PyDict>(py).ok()
    |         ------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         returns a value referencing data owned by the current function
    |         temporary value created here
```

I'm not sure what you had in mind, or how best to implement it, some guidance would be appreciated.

Reply:
I could create a method on `Input` that accepts a callback that would be called on the `Option<&PyDict>`, is that what you had in mind?